flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -605,6 +659,56 @@ private static void commitChanges(
     transaction.commitTransaction();
   }
 
+  private static void commitChanges(
+      Table table,
+      String setLocation,
+      String setSnapshotId,
+      String pickSnapshotId,
+      List<TableChange> schemaChanges,
+      List<TableChange> propertyChanges) {
+    commitChanges(table, setSnapshotId, pickSnapshotId);
+
+    Transaction transaction = table.newTransaction();
+
+    if (setLocation != null) {
+      transaction.updateLocation().setLocation(setLocation).commit();
+    }
+
+    if (!schemaChanges.isEmpty()) {
+      UpdateSchema updateSchema = transaction.updateSchema();
+      FlinkAlterTableUtil.applySchemaChanges(updateSchema, schemaChanges);
+      updateSchema.commit();
+    }
+
+    if (!propertyChanges.isEmpty()) {
+      UpdateProperties updateProperties = transaction.updateProperties();
+      FlinkAlterTableUtil.applyPropertyChanges(updateProperties, propertyChanges);
+      updateProperties.commit();
+    }
+
+    transaction.commitTransaction();
+  }
+
+  private static void commitChanges(Table table, String setSnapshotId, String pickSnapshotId) {
```

Comment:
this method name is a little misleading and confusing. maybe `applyManageSnapshots` would be more intuitive.

also curious why this method doesn't live inside `FlinkAlterTableUtil`?

Reply:
Maybe `commitManageSnapshots`?